src/clj/puppetlabs/services/legacy_routes/legacy_routes_core.clj

```diff
@@ -0,0 +1,120 @@
+                     :query-string (ring-codec/form-encode
+                                     (merge
+                                       query-params
+                                       (when environment {:environment environment}))))
+              (dissoc :params :route-params))]
+      (handler compat-request))))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Routing
+
+(defn v2_0-routes
+  "Creates the compojure routes to handle the 3.X master's '/v2.0' routes."
+  [request-handler]
+  (compojure/routes
+    (compojure/GET "/environments" request
+                   (request-handler request))))
+
+(defn legacy-master-routes
+  [master-request-handler ca-request-handler]
+  (compojure/routes
+    (compojure/GET "/node/*" request (master-request-handler request))
+    (compojure/GET "/facts/*" request (master-request-handler request))
+    (compojure/GET "/file_content/*" request (master-request-handler request))
+    (compojure/GET "/file_metadatas/*" request (master-request-handler request))
+    (compojure/GET "/file_metadata/*" request (master-request-handler request))
+    (compojure/GET "/file_bucket_file/*" request (master-request-handler request))
+    ;; Coercing Content-Type to "application/octet-stream" for Puppet 4
+    ;; compatibility.
+    (compojure/PUT "/file_bucket_file/*" request
+      (master-request-handler
+        (-> request
+          (assoc :content-type "application/octet-stream") ; deprecated in ring >= 1.2
+          (assoc-in [:headers "content-type"] "application/octet-stream")))) ; ring >= 1.2 spec
+    (compojure/HEAD "/file_bucket_file/*" request (master-request-handler request))
+    (compojure/GET "/catalog/*" request (master-request-handler request))
+    (compojure/POST "/catalog/*" request (master-request-handler request))
+    (compojure/PUT "/report/*" request (master-request-handler request))
+    (compojure/GET "/resource_type/*" request (master-request-handler request))
+    (compojure/GET "/resource_types/*" request (master-request-handler request))
+    (compojure/GET "/status/*" request (master-request-handler request))
+
+    ;; TODO: when we get rid of the legacy dashboard after 3.4, we should remove
+    ;; this endpoint as well.  It makes more sense for this type of query to be
+    ;; directed to PuppetDB.
+    (compojure/GET "/facts_search/*" request (master-request-handler request))
+
+    ;; legacy CA routes
+    (compojure/ANY "/certificate_status/*" request (ca-request-handler request))
+    (compojure/ANY "/certificate_statuses/*" request (ca-request-handler request))
+    (compojure/GET "/certificate/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_revocation_list/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_request/*" request (ca-request-handler request))
+    (compojure/GET "/certificate_requests/*" request (ca-request-handler request))))
```

Comment:
We explicitly didn't support this in the CA rewrite because nothing uses it - it existed only because of the magical plurality+indirector nonsense.  Plus, it does not offer any information that isn't available from the other endpoints.

Puppet Server has never supported it, you would just get a 404 Not Found.  Probably best to keep it that way and not introduce support for it now?